### ISSUE TITLE
When I use model inference, why do the embedding generated by the same sentence are different every time

### ISSUE THREAD

**Issue by A6Matrix** [ckiplab/ckip-transformers#13]
this is my code 
![image](https://user-images.githubusercontent.com/18898570/131991716-56e7a5d1-0ca5-4644-938b-91ed63b401c2.png)
First generated embedding ：
 ![PS0@4H0IGM_(97GGHJ FGEB](https://user-images.githubusercontent.com/18898570/131992100-ea3d9f77-15ea-4909-b215-ae272c0f3a1d.png)
Second generated embedding ：
![~VOU(8DDLI)7(TDZCPPHDU4](https://user-images.githubusercontent.com/18898570/131992413-06364771-fdb3-48af-afb6-1550aadfe578.png)
You can see that the sentence embedding generated twice are different

**Comment by qqaatw** [comment]
Could you paste your code directly on your post instead of images so we can easily reproduce? thanks.

**Comment by A6Matrix** [comment]
> Could you paste your code directly on your post instead of images so we can easily reproduce? thanks.

`import torch
from transformers import (BertTokenizerFast,AutoModel,)

tokenizer = BertTokenizerFast.from_pretrained('bert-base-chinese')
albert_model = AutoModel.from_pretrained('ckiplab/albert-tiny-chinese')

def encode_text(text):
    text_code = tokenizer(text,padding=True,truncation=True,return_tensors='pt')
    input_ids = text_code['input_ids']
    attention_mask = text_code['attention_mask']
    token_type_ids = text_code['token_type_ids']
    print('input_ids',input_ids)
    print('attention_mask',attention_mask)
    print('token_type_ids',token_type_ids)
    with torch.no_grad():
        output = albert_model(input_ids,attention_mask=attention_mask,token_type_ids=token_type_ids)
        embed = output.pooler_output
    return embed

text = '聪明孩子'
text_embed = encode_text(text)
print(text_embed)`

**Comment by A6Matrix** [comment]
```Python
import torch
from transformers import (BertTokenizerFast,AutoModel,)

tokenizer = BertTokenizerFast.from_pretrained('bert-base-chinese')
albert_model = AutoModel.from_pretrained('ckiplab/albert-tiny-chinese')

def encode_text(text):
    text_code = tokenizer(text,padding=True,truncation=True,return_tensors='pt')
    input_ids = text_code['input_ids']
    attention_mask = text_code['attention_mask']
    token_type_ids = text_code['token_type_ids']
    print('input_ids',input_ids)
    print('attention_mask',attention_mask)
    print('token_type_ids',token_type_ids)
    with torch.no_grad():
        output = albert_model(input_ids,attention_mask=attention_mask,token_type_ids=token_type_ids)
        embed = output.pooler_output
    return embed

text = '聪明孩子'
text_embed = encode_text(text)
print(text_embed)
```

**Comment by A6Matrix** [comment]
> Could you paste your code directly on your post instead of images so we can easily reproduce? thanks.
this is code
```Python
import torch
from transformers import (BertTokenizerFast,AutoModel,)

tokenizer = BertTokenizerFast.from_pretrained('bert-base-chinese')
albert_model = AutoModel.from_pretrained('ckiplab/albert-tiny-chinese')

def encode_text(text):
    text_code = tokenizer(text,padding=True,truncation=True,return_tensors='pt')
    input_ids = text_code['input_ids']
    attention_mask = text_code['attention_mask']
    token_type_ids = text_code['token_type_ids']
    print('input_ids',input_ids)
    print('attention_mask',attention_mask)
    print('token_type_ids',token_type_ids)
    with torch.no_grad():
        output = albert_model(input_ids,attention_mask=attention_mask,token_type_ids=token_type_ids)
        embed = output.pooler_output
    return embed

text = '聪明孩子'
text_embed = encode_text(text)
print(text_embed)
```

**Comment by A6Matrix** [comment]
> Could you paste your code directly on your post instead of images so we can easily reproduce? thanks.

```Python
import torch
from transformers import (BertTokenizerFast,AutoModel,)

tokenizer = BertTokenizerFast.from_pretrained('bert-base-chinese')
albert_model = AutoModel.from_pretrained('ckiplab/albert-tiny-chinese')

def encode_text(text):
    text_code = tokenizer(text,padding=True,truncation=True,return_tensors='pt')
    input_ids = text_code['input_ids']
    attention_mask = text_code['attention_mask']
    token_type_ids = text_code['token_type_ids']
    print('input_ids',input_ids)
    print('attention_mask',attention_mask)
    print('token_type_ids',token_type_ids)
    with torch.no_grad():
        output = albert_model(input_ids,attention_mask=attention_mask,token_type_ids=token_type_ids)
        embed = output.pooler_output
    return embed

text = '聪明孩子'
text_embed = encode_text(text)
print(text_embed)
```

**Comment by qqaatw** [comment]
The actual model behind `ckiplab/albert-tiny-chinese` is `AlbertForMaskedLM`, which has its pooler layer disabled by default, so the weights of the pooler layer will be newly initialized every time when instantiating the model.

If you checkout `output.last_hidden_state`, the output value will remain consistent.

Kindly notice the warning when you instantiate the model:
```
Some weights of the model checkpoint at ckiplab/albert-tiny-chinese were not used when initializing AlbertModel: ['predictions.decoder.bias', 'predictions.dense.weight', 'predictions.LayerNorm.weight', 'predictions.decoder.weight', 'predictions.dense.bias', 'predictions.LayerNorm.bias', 'predictions.bias']
- This IS expected if you are initializing AlbertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing AlbertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of AlbertModel were not initialized from the model checkpoint at ckiplab/albert-tiny-chinese and are newly initialized: ['albert.pooler.weight', 'albert.pooler.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
```

**Comment by A6Matrix** [comment]
> n a do

I found that when I set up the random seed, the pooler layer will not change

**Comment by A6Matrix** [comment]
> The actual model behind `ckiplab/albert-tiny-chinese` is `AlbertForMaskedLM`, which has its pooler layer disabled by default, so the weights of the pooler layer will be newly initialized every time when instantiating the model.
> 
> If you checkout `output.last_hidden_state`, the output value will remain consistent.
> 
> Kindly notice the warning when you instantiate the model:
> 
> ```
> Some weights of the model checkpoint at ckiplab/albert-tiny-chinese were not used when initializing AlbertModel: ['predictions.decoder.bias', 'predictions.dense.weight', 'predictions.LayerNorm.weight', 'predictions.decoder.weight', 'predictions.dense.bias', 'predictions.LayerNorm.bias', 'predictions.bias']
> - This IS expected if you are initializing AlbertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
> - This IS NOT expected if you are initializing AlbertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
> Some weights of AlbertModel were not initialized from the model checkpoint at ckiplab/albert-tiny-chinese and are newly initialized: ['albert.pooler.weight', 'albert.pooler.bias']
> You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
> ```

I found that when I set up the random seed, the pooler layer will not change

**Comment by qqaatw** [comment]
That's the expected behavior as you fixed the seed of random generator, the initialized weights of the pooler layer will remain consistent. However, please note that these weights are still **untrained**, you'll have to train them for your downstream task accordingly.

**Comment by A6Matrix** [comment]
> That's the expected behavior as you fixed the seed of random generator, the initialized weights of the pooler layer will remain consistent. However, please note that these weights are still **untrained**, you'll have to train them for your downstream task accordingly.

OK，thank you

**Comment by qqaatw** [comment]
No problem ;)
